contracts/Remittance.sol

```diff
@@ -1,16 +1,140 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
 
+    struct RemittanceInstance {
+        address funder;
+        uint256 fundsOwed;
+        uint256 expiryDate;
+    }
+
+    event RemittanceFundsCreated(
+        bytes32 indexed hash,
+        bytes32 passwordBrokerHash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount,
+        uint256 expiryDate
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        bytes32 indexed password,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReclaimed(
+        bytes32 indexed hash,
+        bytes32 indexed passwordBrokerHash,
+        address indexed funder,
+        uint256 amount
+    );
+
+    // The funder can create a remittance
+    // Hash is a sha256 hash of the broker address + recipient password
+    function create(bytes32 passwordBrokerHash, address broker, uint256 expiryDate)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        require(broker != msg.sender, "Caller cannot be the broker");
+        require(msg.value > 0, "Amount must be greater than 0");
+        require(broker != address(0), "Address cannot be zero");
+
+        // A 15 min margin has been added to allow for any lag
+        require(expiryDate >= block.timestamp + (24 * 60 * 60) - 1500, "Expiry less than 24h ahead");
+        checkEmptyHash(passwordBrokerHash);
+
+        bytes32 rehash = keccak256(abi.encodePacked(passwordBrokerHash, address(this)));
```

Comment:
I think you are missing a trick here. Do you see it?

Reply:
I think I did see it. I considered the following

- Creating a function to be DRY but thought this might cost more gas.
- Making the hashPasswordBroker() more generic but did not want to change the specificity of the public interface.

I have refactored to a function as the advantage outweigh the gas cost.

Reply:
You created this so as to avoid the situation where a password that works with instance1 will work with instance2, right?
Did you confirm this is the case with a unit test?
I don't think it does because the rehash happens on the inside of each function. It is an _implementation detail_ about where the struct gets stored.

Reply:
I see exactly what you mean. Useless on the inside. Not sure what I was thinking at the time. I was focusing perhaps on the mapping keys and not the visible function arguments. This should be supported and verified with a test.

Reply:
I have tested this simply by trying to call the hashPasswordBroker() in two different contracts using the same password and broker and checking they return different hashes.